pkg/kube/longhorn-utils.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+#
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+LONGHORN_VERSION=v1.6.2
+
+longhorn_install() {
+    node_name=$1
+    logmsg "Installing longhorn version ${LONGHORN_VERSION}"
+    apply_longhorn_disk_config "$node_name"
+    lhCfgPath=/etc/lh-cfg-${LONGHORN_VERSION}.yaml
+    if ! grep -q 'create-default-disk-labeled-nodes: true' "$lhCfgPath"; then
+            sed -i '/  default-setting.yaml: |-/a\    create-default-disk-labeled-nodes: true' "$lhCfgPath"
+    fi
+    if ! kubectl apply -f "$lhCfgPath"; then
+            return 1
+    fi
+    return 0
+}
+
+longhorn_is_ready() {
+    lhStatus=$(kubectl -n longhorn-system get daemonsets -o json | jq '.items[].status | .numberReady==.desiredNumberScheduled' | tr -d '\n')
+    if [ "$lhStatus" != "truetruetrue" ]; then
+            return 1
+    fi
+    return 0
+}
+
+apply_longhorn_disk_config() {
+        node=$1
+        kubectl label node "$node" node.longhorn.io/create-default-disk='config'
+        kubectl annotate node "$node" node.longhorn.io/default-disks-config='[ { "path":"/persist/vault/volumes", "allowScheduling":true }]'
+}
+
+check_overwrite_nsmounter() {
+        ### REMOVE ME+
+        # When https://github.com/longhorn/longhorn/issues/6857 is resolved, remove this 'REMOVE ME' section
```

Comment:
As far as I see the issue is closed...

Reply:
I need to re-engage with that thread, it seems others joined in with somewhat similar symptoms but different root causes and my original issue was mistakenly marked with them.